src/core/view_manager.cpp

```diff
@@ -861,6 +852,56 @@ void ViewManager::redirect_join_attempt(DerechoSST& gmsSST) {
             curr_view->member_ips_and_ports[rank_of_leader]));
 }
 
+void ViewManager::process_new_sockets() {
+    tcp::socket client_socket;
+    {
+        auto pending_new_sockets_locked = pending_new_sockets.locked();
+        client_socket = std::move(pending_new_sockets_locked.access.front());
+        pending_new_sockets_locked.access.pop_front();
+    }
+    dbg_default_info("Background thread got a connection from {}", client_socket.get_remote_ip());
+    //Exchange version codes; close the socket if the client has an incompatible version
+    uint64_t joiner_version_code;
+    client_socket.exchange(my_version_hashcode, joiner_version_code);
+    if(joiner_version_code != my_version_hashcode) {
+        rls_default_warn("Rejected a connection from client at {}. Client was running on an incompatible platform or used an incompatible compiler.",
+                         client_socket.get_remote_ip());
+        return;
+    }
+    JoinRequest join_request;
+    client_socket.read(join_request);
+    if (join_request.is_external) {
+        dbg_default_info("The join request is an external request from {}.", join_request.joiner_id);
+        external_join_handler(client_socket, join_request.joiner_id);
+    } else {
+        if (active_leader) {
+            pending_join_sockets.emplace_back(std::make_pair(join_request.joiner_id, std::move(client_socket)));
+        } else {
+            redirect_join_attempt(client_socket);
+        }
+    }
+    return;
+}
+void ViewManager::external_join_handler(tcp::socket& client_socket, const node_id_t& joiner_id) {
+    if(curr_view->rank_of(joiner_id) != -1) {
+        // external can't have same id as any member
+        client_socket.write(JoinResponse{JoinResponseCode::ID_IN_USE, getConfUInt32(CONF_DERECHO_LOCAL_ID)});
+    }
+    client_socket.write(JoinResponse{JoinResponseCode::OK, getConfUInt32(CONF_DERECHO_LOCAL_ID)});
+    ExternalClientRequest request;
+    client_socket.read(request);
+    if (request == ExternalClientRequest::GET_VIEW) {
+        dbg_default_debug("Background thread got an external client connection from {}", client_socket.get_remote_ip());
```

Comment:
This message is also redundant, and doesn't really make sense. Did you mean to report that an external client requested a view?